pages/dashboard/v2/details/index.js

```diff
@@ -0,0 +1,30 @@
+import Navbar from '../../../../components/navbar';
+import Layout from '../../../../components/layout';
+import Head from 'next/head';
+import Link from 'next/link';
+import { useRouter } from 'next/router';
+
+export default function Details() {
+  const router = useRouter();
+  const data = router.query;
+  return (
+    <Layout>
+      <Head>
+        <title>Student Details</title>
+        <meta name='description' content='Generated by create next app' />
+        <link rel='icon' href='/favicon.ico' />
+      </Head>
+      <Navbar>
+        <div className='border-solid border-2 pl-4 pr-4'>
+          <Link href={'/classes'}>Classes</Link>
+        </div>
+        <div className='border-solid border-2 pl-4 pr-4'>
+          <Link href={'/'}> Menu</Link>
+        </div>
+      </Navbar>
+      <div>
+        <h1>{`${data.studentName}'s Progress in: ${data.classroomName} `}</h1>
+      </div>
+    </Layout>
+  );
+}
```

Comment:
As mentioned elsewhere in this code review, in order to make use of Next.js' dynamic routing we need to preserve the following directory structure for the details folder:

details/[id]/[studentName]/index.js.

So delete the there should be a folder under details named [id], then add a folder in  there titled [studentName] and in this folder is where you would add the index.js file.

your index.js file should look something like the following:

`import Head from 'next/head';
import Navbar from '../../../../../../components/navbar';
import Link from 'next/link';
import { getSession } from 'next-auth/react';
import { useRouter } from 'next/router';

export async function getServerSideProps(context) {
  const userSession = await getSession(context);
  if (!userSession) {
    context.res.writeHead(302, { Location: '/' });
    context.res.end();
    return {};
  }
  return {
    props: {
      userSession
    }
  };
}

export default function Details({
  userSession
}) {
  const router = useRouter();
  const data = router.query;
  const studentName = data.studentName;
  const classroomName = data.classroomName;
  return (
    <>
    <Head>
        <title>Create Next App</title>
        <meta name='description' content='Generated by create next app' />
        <link rel='icon' href='/favicon.ico' />
      </Head>
      {userSession && (
        <>
          <Navbar>
            <div className='border-solid border-2 pl-4 pr-4'>
              <Link href={'/classes'}>Classes</Link>
            </div>
            <div className='border-solid border-2 pl-4 pr-4'>
              <Link href={'/'}> Menu</Link>
            </div>
          </Navbar>
          <h1>{studentName}&apos; progress in {classroomName}</h1>
        </>
      )}
    </>
  );
}
`